super-austin/drug-reaction-reporter
# Backend: Implement API Endpoint for Drug Reactions

# Requirements

Create an API endpoint. The input should be the name of a drug and the output should be a list of reactions along with the number of times the reaction was reported. 

The API should use the FDA’s adverse reaction dataset (https://open.fda.gov/apis/drug/event ) to get the data it needs. Use the **patient.drug.openfda.brand_name** field for querying adverse reactions to a drug ("Tylenol" or "Vimizim" for testing). 

# Main Tech-Stack

- Primary: Express.js
- Extra: axios, cors, dotenv
- Package Manager: NPM

# How to Test

- Running the server

```
npm run dev
```
- The backend server would be hosted on `http://localhost:5000`.
- The `/health` endpoint returns an appropriate success message.
- Manual testing of the `/api/drug-reactions` endpoint with various drug names (e.g., "Tylenol") to ensure correct responses.

# Screenshots

![image](https://github.com/user-attachments/assets/f5abed0a-a676-496f-b616-c8c6a55e78a4)
